tests/results/test_result_factory.py

```diff
@@ -896,3 +896,45 @@ def my_flow():
 
     result_factory = my_flow()
     await _verify_default_storage_creation_without_persistence(result_factory)
+
+
+@pytest.mark.parametrize(
+    "options,expected",
+    [
+        (
+            {
+                "persist_result": True,
+                "cache_result_in_memory": False,
+                "result_serializer": "json",
+            },
+            {
+                "persist_result": True,
+                "cache_result_in_memory": False,
+                "serializer": JSONSerializer(),
+            },
+        ),
+        (
+            {
+                "persist_result": False,
+                "cache_result_in_memory": True,
+                "result_serializer": "json",
+            },
+            {
+                "persist_result": False,
+                "cache_result_in_memory": True,
+                "serializer": JSONSerializer(),
+            },
+        ),
+    ],
+)
+async def test_result_factory_from_task_with_no_flow_run_context(options, expected):
+    @task(**options)
+    def my_task():
+        pass
+
+    result_factory = await ResultFactory.from_task(task=my_task)
```

Comment:
Love it, just one oddball question, though: is there an assertion we can make in this test to show that there's no current flow run context?  Like `assert not FlowRunContext.get()`?  I like to assert my preconditions if at all possible, just in case things change in the future.

Reply:
definitely, i like it! will add now 👍

Reply:
added in https://github.com/PrefectHQ/prefect/pull/11134/commits/dc5c0e31a4cd5a004f62cfe66331172e1ec67f47